radix-engine/src/transaction/transaction_executor.rs

```diff
@@ -187,7 +188,7 @@ where
             crate::kernel::resources_tracker::ResourcesTracker::start_measurement();
 
         // Prepare
-        let mut track = Track::<_, JmtMapper>::new(self.substate_db);
+        let mut track = Track::<_, SpreadPrefixKeyMapper>::new(self.substate_db);
```

Comment:
Maybe this is the answer inside the engine - we make Track a trait - and we instantiate an implementation of `Track::<_, SpreadPrefixKeyMapper>` in the transaction executor when we create it - and then things like `to_state_updates` are functions on the track interface, rather than requiring the mapper to be provided.

Track then save a `TypedSubstateStore<DatabaseKeyMapper>` in its struct so that it knows _which_ typed substate store it is, and so we don't have to specify at each call-site.

That would at least get rid of most of the `SpreadPrefixKeyMapper` generics around the `radix-engine` codebase.

Reply:
(I imagine we can do a similar trick in `scrypto-unit` and `radix-engine-queries`)

Reply:
and [the 9th time](https://github.com/radixdlt/radixdlt-scrypto/pull/986#discussion_r1181195620).